WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift

```diff
@@ -244,15 +218,69 @@ private extension ProductFormViewController {
         moreDetailsContainerView.setContentCompressionResistancePriority(.required, for: .vertical)
         moreDetailsContainerView.setContentHuggingPriority(.required, for: .vertical)
 
-        updateMoreDetailsButtonVisibility(product: product)
+        updateMoreDetailsButtonVisibility()
+    }
+}
+
+private extension ProductFormViewController {
+    func observeProduct() {
+        cancellableProduct = viewModel.observableProduct.subscribe { [weak self] product in
+            self?.onProductUpdated(product: product)
+        }
+    }
+
+    func observeProductName() {
+        cancellableProductName = viewModel.productName.subscribe { [weak self] name in
+            self?.updateNavigationBarTitle(productName: name)
+        }
+    }
+
+    func observeUpdateCTAVisibility() {
+        cancellableUpdateEnabled = viewModel.isUpdateEnabled.subscribe { [weak self] isUpdateEnabled in
+            self?.updateNavigationBar(isUpdateEnabled: isUpdateEnabled)
+        }
+    }
+
```

Comment:
wow @pmusolino really nice catch! this issue was caused by the `isUpdateEnabled` isn't triggered by the password change because the value remains the same after the remote update (it was only triggered when the value changes).

In https://github.com/woocommerce/woocommerce-ios/pull/2348/commits/67eaf903bdcc5fd0dea209d63002845e3f5f22fe, `isUpdateEnabled` is now triggered when the password is reset (e.g. after a remote fetch or update).

In the case of remote Product updates, the Product's `dateModified` changes and thus the `isUpdateEnabled` is triggered in Product's `didSet`. This is not the case for password, because the password theoretically would be the same before and after a remote update.